pkg/cluster/cluster.go

```diff
@@ -138,6 +142,13 @@ func (c *Cluster) setup() error {
 		return fmt.Errorf("unexpected cluster phase: %s", c.status.Phase)
 	}
 
+	if clustertls.IsSecureClient(c.cluster.Spec) {
```

Comment:
IsSecureClient should belong to pkg cluster i feel. it has nothing to do with tls in general.